standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -4393,4 +4393,27 @@ ReplicationMetricList getReplicationMetrics(GetReplicationMetricsRequest
 
   AbortCompactResponse abortCompactions(AbortCompactionRequest request) throws TException;
 
+  /**
+   * Sets properties.
+   * @param nameSpace the property store namespace
+   * @param properties a map keyed by property path mapped to property values
+   * @return true if successful, false otherwise
+   * @throws TException
+   */
+  default boolean setProperties(String nameSpace, Map<String, String> properties) throws TException {
+    throw new UnsupportedOperationException();
```

Comment:
I think this definition is incorrect. We should only declare the method here and any other dummy clients or clients that don't support this method should define this as unsupported.
Currently, SessionHiveMetaStoreClient would throw this exception. Can you please implement these methods in SessionHiveMetastoreClient as well?

Reply:
SessionHiveMetaStoreClient extends HiveMetaStoreClient where the default methods are overridden and implemented.